src/selectors/entities/channels.js

```diff
@@ -503,7 +504,7 @@ export const getSortedFavoriteChannelWithUnreadsIds = createIdsSelector(
 
             const channel = channels[id];
             const otherUserId = getUserIdFromChannelName(currentUser.id, channel.name);
-            if (channel.type === General.DM_CHANNEL && !isDirectChannelVisible(profiles[otherUserId] || otherUserId, config, prefs, channel)) {
+            if (channel.type === General.DM_CHANNEL && !isDirectChannelVisible(profiles[otherUserId] || otherUserId, config, prefs, channel, currentChannelId)) {
```

Comment:
I think you are adding the channelId at the end, but is expected lastPost and isUnread in the middle, maybe you have add `null`, `null`, `currentChannelId`